fed_test_utils/fed.go

```diff
@@ -153,51 +154,89 @@ func NewFedTest(t *testing.T, originConfig string) (ft *FedTest) {
 	directorStartTime := time.Now().Add(-6 * time.Minute)
 	director.SetStartupTime(directorStartTime)
 
-	config.InitConfig()
 	err = config.InitServer(ctx, modules)
 	require.NoError(t, err)
 
 	// Read in any config we may have set
+	var originConf interface{}
 	if originConfig != "" {
 		viper.SetConfigType("yaml")
 		err = viper.MergeConfig(strings.NewReader(originConfig))
 		require.NoError(t, err, "error reading config")
+
+		err := yaml.Unmarshal([]byte(originConfig), &originConf)
+		require.NoError(t, err, "error unmarshalling into interface")
+
+		confMap := originConf.(map[string]interface{})
+
+		originRaw, exists := confMap["Origin"]
+
+		if exists {
+			originMap := originRaw.(map[string]interface{})
+
+			// Override the test directory from the config file with our temp directory
+			if exportsRaw, ok := originMap["Exports"]; ok {
+				for i, item := range exportsRaw.([]interface{}) {
+					originDir, err := os.MkdirTemp("", fmt.Sprintf("Export%d", i))
+					assert.NoError(t, err)
+					t.Cleanup(func() {
+						err := os.RemoveAll(originDir)
+						require.NoError(t, err)
+					})
+
+					exportMap := item.(map[string]interface{})
+					exportMap["StoragePrefix"] = originDir
+
+					// Change the permissions of the temporary origin directory
+					permissions = os.FileMode(0755)
+					err = os.Chmod(originDir, permissions)
+					require.NoError(t, err)
+
+					// Change ownership on the temporary origin directory so files can be uploaded
+					uinfo, err := config.GetDaemonUserInfo()
+					require.NoError(t, err)
+					require.NoError(t, os.Chown(originDir, uinfo.Uid, uinfo.Gid))
+
+					// Start off with a Hello World file we can use for testing in each of our exports
+					err = os.WriteFile(filepath.Join(originDir, "hello_world.txt"), []byte("Hello, World!"), os.FileMode(0644))
+					require.NoError(t, err)
+				}
+			} else {
+				originDir, err := os.MkdirTemp("", fmt.Sprintf("Export%s", "test"))
+				assert.NoError(t, err)
+				t.Cleanup(func() {
+					err := os.RemoveAll(originDir)
+					require.NoError(t, err)
+				})
+
+				originMap["StoragePrefix"] = originDir
+
+				permissions = os.FileMode(0755)
+				err = os.Chmod(originDir, permissions)
+				require.NoError(t, err)
+
+				// Change ownership on the temporary origin directory so files can be uploaded
+				uinfo, err := config.GetDaemonUserInfo()
+				require.NoError(t, err)
+				require.NoError(t, os.Chown(originDir, uinfo.Uid, uinfo.Gid))
+
+				// Start off with a Hello World file we can use for testing in the StoragePrefix
+				err = os.WriteFile(filepath.Join(originDir, "hello_world.txt"), []byte("Hello, World!"), os.FileMode(0644))
+				require.NoError(t, err)
+			}
+		}
```

Comment:
The majority of the code in the `if exportsRaw, ok := originMap["Exports"]; ok {` is duplicated in the subsequent `else` -- can you turn some of it into an inline helper function somewhere?